aiokafka/protocol/message.py

```diff
@@ -47,7 +51,40 @@ class Message(Struct):
         22  # crc(4), magic(1), attributes(1), timestamp(8), key+value size(4*2)
     )
 
-    def __init__(self, value, key=None, magic=0, attributes=0, crc=0, timestamp=None):
+    @overload
+    def __init__(
+        self,
+        *,
+        value: Optional[bytes],
+        key: Optional[bytes] = None,
+        magic: Literal[0],
+        attributes: int = 0,
+        crc: int = 0,
+        timestamp: None = None,
+    ) -> None: ...
+
+    @overload
+    def __init__(
+        self,
+        *,
+        value: Optional[bytes],
+        key: Optional[bytes] = None,
+        magic: Literal[1],
+        attributes: int = 0,
+        crc: int = 0,
+        timestamp: Optional[int] = None,
+    ) -> None: ...
+
+    def __init__(
+        self,
+        *,
+        value: Optional[bytes],
+        key: Optional[bytes] = None,
+        magic: Literal[0, 1] = 0,
+        attributes: int = 0,
+        crc: int = 0,
```

Comment:
Do we need default values for them? I'm especially uncomfortable seeing it for `crc`

Reply:
I think no. I can remove all default values, except timestamp.